### ISSUE TITLE
Sprite clipping with vanilla rendering

### ISSUE THREAD

**Issue by nstlaurent** [Helion-Engine/Helion#907]
Eviternity MAP18

This happens when a sprite is clipped far into a wall. Since the depth buffer is cleared for flats it renders through.

![helion_20241109_03_08_22_0471](https://github.com/user-attachments/assets/2b0d8091-9f1a-40ad-a786-d52b353ae4da)

Also worth noting that software rendering also has this problem.
![image](https://github.com/user-attachments/assets/7225dfb0-7c58-48cf-981c-cca313275ca9)

**Comment by JustinWayland** [comment]
To me, this is a good indication that the map was either wittingly or unwittingly designed with hardware rendering in mind.

I noticed this bug on MAP01 of Valiant when vanilla rendering emulation was enabled:
![Screenshot_20241228_130337](https://github.com/user-attachments/assets/44ea211e-5d34-42b0-8495-4715848c8c3d)

Notably, when vanilla rendering emulation is _disabled_ the bug disappears:
![Screenshot_20241228_131215](https://github.com/user-attachments/assets/790f451c-525d-40b0-935f-07b24f47a563)
(Not exactly the same orientation but disabling vanilla rendering emulation required a restart to take effect, and of course the bug disappears when vanilla rendering is disabled so I couldn't find the exact same orientation again.)

**Comment by nstlaurent** [comment]
The issue happens in vanilla, sometimes the same and sometimes less. Depends on the BSP rendering order. Looks like this is a case where it happens less because there is a subsector slice that renders after the tree sprite covering it a little bit more. This is the worst I could get to happen in dsda software.

![image](https://github.com/user-attachments/assets/5223b3d6-bc22-4d52-993a-64633122315b)

I have been thinking about this and I understand why the original software renderer didn't solve this. It's pretty complex and would be fairly computationally expensive to try to solve. The other issue is any solution here is likely to cause other incorrect sprite rendering issues.

**Comment by JustinWayland** [comment]
Yeah, the best solution is to say "if you encounter clipping in a map, save, set vanilla rendering to false, and restart Helion." This is a map bug, not an engine bug.

Should we close this issue as the GitHub equivalent of WONTFIX?
